astropy/io/fits/hdu/image.py

```diff
@@ -1056,7 +1057,9 @@ def _getdata(self, keys):
                 break
             # This should always break at some point if _getdata is called.
 
-        data = [self[keys[:idx] + (k,) + keys[idx + 1 :]] for k in ks]
+        with ThreadPoolExecutor() as pool:
```

Comment:
Is it wise to use this without giving users an option to control `max_workers`?

Reply:
> Do we also need an option to disable it altogether, just in case?

Maybe it should only be enabled when the user is using `fsspec`?

> Is it wise to use this without giving users an option to control max_workers?

I'm not sure. I'm not sure what the consequences of that could be. Would the option have to go in fits.open?  There are a lot of options already, there. Maybe it could be an environment variable. That would allow the implementation to avoid having to pass the param all the way down into the Section class.

The default threads that Python uses for concurrent.futures threadpool is `min(32, (os.cpu_count() or 1) + 4)`.

https://github.com/python/cpython/blob/main/Lib/concurrent/futures/thread.py#L145